commit 402871743dd33ee17af7fb52f5b37852a6327f3f
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2019-06-14T13:46:47+02:00 (4 months ago)
    
    TRITON-1740 Sizing up flexible disk throws error but works nonetheless

diff --git a/lib/do_instance/do_disk/do_resize.js b/lib/do_instance/do_disk/do_resize.js
index 6bf914c..3b849a0 100644
--- a/lib/do_instance/do_disk/do_resize.js
+++ b/lib/do_instance/do_disk/do_resize.js
@@ -67,6 +67,7 @@ function do_resize(subcmd, opts, args, cb) {
             if (size < ctx.disk.size && !opts.dangerous_allow_shrink) {
                 next(new Error('--dangerous-allow-shrink must be specified ' +
                                 'when shrinking a disk'));
+                return;
             }
 
             cli.tritonapi.resizeInstanceDisk({
@@ -74,9 +75,9 @@ function do_resize(subcmd, opts, args, cb) {
                 diskId: diskId,
                 size: size,
                 dangerousAllowShrink: opts.dangerous_allow_shrink
-            }, function onResized(err, res) {
+            }, function onResized(err, _, res) {
                 if (err) {
-                    cb(err);
+                    next(err);
                     return;
                 }
 
@@ -111,9 +112,10 @@ function do_resize(subcmd, opts, args, cb) {
                     console.log('Resized disk "%s" in %s', disk.id,
                                 common.humanDurationFromMs(duration));
                     next();
-                } else {
-                    next(new Error('Failed to resize disk'));
+                    return;
                 }
+
+                next(new Error('Failed to resize disk'));
             });
         }
     ]}, cb);
